tests/route/test_route_flow_counter.py

```diff
@@ -0,0 +1,158 @@
+        prefix_list = ['1.1.1.0/24', '1.1.2.0/24', '1.1.3.0/24']
+        nexthop_addr = self._get_nexthop(duthost, False)
+        flow_counter_utils.set_route_flow_counter_status(duthost, True)
+        expect_route_flow_counter = len(prefix_list) - 1
+        flow_counter_utils.set_route_flow_counter_pattern(duthost, route_pattern, max_match_count=expect_route_flow_counter)
+        logger.info('Adding {} static routes while max allowed match count is {}'.format(len(prefix_list), expect_route_flow_counter))
+        for prefix in prefix_list:
+            duthost.shell("sonic-db-cli CONFIG_DB hmset 'STATIC_ROUTE|{}' nexthop {}".format(prefix, nexthop_addr))
+
+        logger.info('Verify there are {} route flow counters'.format(expect_route_flow_counter))
+        stats = flow_counter_utils.parse_route_flow_counter_stats(duthost)
+
+        pytest_assert(len(stats) == expect_route_flow_counter, 'Expecetd {} route flow counters, but got {} counters'.format(expect_route_flow_counter, len(stats)))
+
+        logger.info('Removing a route, verify there are still {} route flow counters'.format(expect_route_flow_counter))
+        duthost.shell("sonic-db-cli CONFIG_DB del 'STATIC_ROUTE|{}'".format(prefix_list[0]), module_ignore_errors=True)
+        stats = flow_counter_utils.parse_route_flow_counter_stats(duthost)
+        pytest_assert(len(stats) == expect_route_flow_counter, 'Max allowed match counter is {}, but got {} counters'.format(expect_route_flow_counter, len(stats)))
+
+        expect_route_flow_counter -= 1
+        logger.info('Set max_match_count to {}, verify there is {} route flow counters'.format(expect_route_flow_counter, expect_route_flow_counter))
+        flow_counter_utils.set_route_flow_counter_pattern(duthost, route_pattern, max_match_count=expect_route_flow_counter)
+        stats = flow_counter_utils.parse_route_flow_counter_stats(duthost)
+        pytest_assert(len(stats) == expect_route_flow_counter, 'Max allowed match counter is {}, but got {} counters'.format(expect_route_flow_counter, len(stats)))
+
+        expect_route_flow_counter += 1
+        logger.info('Set max_match_count to {}, verify there is {} route flow counters'.format(expect_route_flow_counter, expect_route_flow_counter))
+        flow_counter_utils.set_route_flow_counter_pattern(duthost, route_pattern, max_match_count=expect_route_flow_counter)
+        stats = flow_counter_utils.parse_route_flow_counter_stats(duthost)
+        pytest_assert(len(stats) == expect_route_flow_counter, 'Max allowed match counter is {}, but got {} counters'.format(expect_route_flow_counter, len(stats)))
+
+        for prefix in prefix_list:
+            duthost.shell("sonic-db-cli CONFIG_DB del 'STATIC_ROUTE|{}'".format(prefix), module_ignore_errors=True)
+        flow_counter_utils.set_route_flow_counter_status(duthost, False)
+        flow_counter_utils.remove_route_flow_counter_pattern(duthost, route_pattern)
+
+    def _get_nexthop(self, duthost, ipv6):
```

Comment:
Why is this a private function?

Reply:
because no one will use this function outside this class. what is the problem here?